session/src/main/java/com/tapptitude/session/SessionManagerImpl.kt

```diff
@@ -0,0 +1,47 @@
+package com.tapptitude.session
+
+import com.tapptitude.session.model.LoggedIn
+import com.tapptitude.session.model.LoggedOut
+import com.tapptitude.session.model.LoginState
+import com.tapptitude.session.storage.LocalSessionDataManager
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.launch
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.inject
+
+internal class SessionManagerImpl : SessionManager, KoinComponent {
+    private val localSessionDataManager: LocalSessionDataManager by inject()
```

Comment:
Wouldn't it be better to have localSessionDataManager as a contructor injection?

Reply:
Great catch ;)